utils/zerovec/derive/src/ule.rs

```diff
@@ -278,10 +299,8 @@ fn make_ule_enum_impl(
 }
```

Comment:
Nit (line 280): Clippy doesn't like functions that return an `()` Error type. Should we consider a more expressive error type here?

Reply:
Now that we decided to keep the TryFrom impl, we should really consider using a more expressive error type here. Could be just ZeroVecError, or could be a custom error type just for this one case, like KeyError

Reply:
Overall I'm personally not a huge fan of that lint because yes, it is better to have a different error type, but in many cases it feels like overkill.

Reply:
Well, I would argue that this is not really "just an internal error." It shows up on TryFrom impls for public enums, which are impls that are generally useful, to the extent that I can see them being used in client code.

If you really don't want to introduce a ZST for this error, then return Option with must_use.

Reply:
Yeah that'd work!